File: Python/reverse-linked-list.py
# Definition for singly-linked list.
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        result:Optional[ListNode] = ListNode(-1)
        runner:Optional[ListNode] = head
        while(runner is not None):
            node=runner
            runner=runner.next
            node.next=None
            next: Optional[ListNode] = result.next
            result.next = node
            node.next = next
        return result.next
        
            
            
            

File: Java/reverse-linked-list.java

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
     ListNode(int val) { this.val = val; }
     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}
 
class Solution {
    public ListNode reverseList(ListNode head) {
      if(head ==null){
        return head;
      }
      ListNode runner=head.next;
      ListNode next=null;
      ListNode pred=head;
      pred.next=null;
      while (runner!=null) {
        next=runner.next;
        runner.next=pred;
        pred=runner;
        runner=next;
      }
      return pred;
    }
}
